### PR TITLE
Feature/signup

### LINKED ISSUE #2: feat: user should be able to signup as a patient
Create an endpoint that takes user details including:
- first name
- last name
- email
- password
- phone number
- preexisting condition(s)
- image

Acceptance criteria: 
- Thorough input validation
- Return relevant success and error mesasge
- Write integration tests
- Include a screenshot of postman request in pull request

### PR DESCRIPTION
**Pull request by John-Akins**
## Description
Rest API endpoint for signup,
Nodejs express framework was used,
PostgreSQL was used for database,
Sequelize was used as ORM 
Mocha was used to setup Test Suite.

Completes #2  

## How Has This Been Tested?
1.  Clone/Pull/Fetch this branch.
2.  Run `npm install` to install the dependencies.
3.  Run `npm test-a` to run the tests.
4.  Test the `/api/v1/signup` endpoint of the server with postman

## Screenshots

Postman request

request

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have added necessary inline code documentation
-   [x] I have added tests that prove my fix is effective and that this feature works
-   [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/controllers/AuthController.js [hunk 1]
```diff
@@ -29,28 +26,50 @@ const userToken = (user) => {
...
+      password = hashPassword(password);
+
+      if (!req.file) {
+        return sendSuccessResponse(res, 422, 'Please select an image file to upload');
+      }
+
+      const image = imageUploader(req.file);
+
+      const user = {
+        firstName, surName, lastName, email, password, phone, conditions, image, role
```
Comment (adaezeodurukwe) [inline src/controllers/AuthController.js]: firstname, surname and lastname? please check this

File: src/controllers/AuthController.js [hunk 2]
```diff
@@ -29,28 +26,50 @@ const userToken = (user) => {
...
+  async signup(req, res) {
+    try {
+      const {
+        firstName, surName, lastName, email, phone, conditions, role
+      } = req.body;
+      let { password } = req.body;
+      password = hashPassword(password);
+
+      if (!req.file) {
+        return sendSuccessResponse(res, 422, 'Please select an image file to upload');
```
Comment (adaezeodurukwe) [inline src/controllers/AuthController.js]: For now, no image will be required on sign up, for better user experience. profiles will be completed on user dashboards.

File: src/database/migrations/20200506192302-create-user-table.js [hunk 3]
```diff
@@ -7,15 +7,15 @@ module.exports = {
...
-    surname: {
+    firstName: {
       allowNull: false,
       type: Sequelize.STRING
     },
-    middleName: {
+    lastName: {
       allowNull: true,
       type: Sequelize.STRING
     },
```
Comment (adaezeodurukwe) [inline src/database/migrations/20200506192302-create-user-table.js]: also, surname and last name mean the same thing so let's eliminate one

File: src/routes/auth.js [hunk 4]
```diff
@@ -1,9 +1,12 @@
-/* eslint-disable no-unused-vars */
 import express from 'express';
 import AuthController from '../controllers/AuthController';
-import Auth from '../middlewares/Auth';
+// import Auth from '../middlewares/Auth';
```
Comment (adaezeodurukwe) [inline src/routes/auth.js:3]: kindly remove commented code

File: src/database/migrations/20200506192302-create-user-table.js (hunk 3 above)
Comment (John-Akins) [reply src/database/migrations/20200506192302-create-user-table.js]: Okay, noted

File: src/controllers/AuthController.js (hunk 2 above)
Comment (John-Akins) [reply src/controllers/AuthController.js]: It was part of the fields in the spec you assigned me. It's why you're seeing it. But you're right, it's not necessary on sign-up.

File: src/controllers/AuthController.js (hunk 1 above)
Comment (John-Akins) [reply src/controllers/AuthController.js]: Noted

**Review by adaezeodurukwe — COMMENTED** [review]
Also return created user details in the response body, especially the email.

**Comment by lovisgod** [comment]
Please make the review pass.

File: src/controllers/AuthController.js (hunk 1 above)
Comment (lovisgod) [reply src/controllers/AuthController.js]: I think this is in code to write and not what the user is to see.
Also, it is save to use a camel case for two words that are together.

**Comment by lovisgod** [comment]
I don't think returning user details on signup is cool. We are prompting the user to verify their account from there mail. So sending these details back is of no necessity. @adaezeodurukwe

**Review by lovisgod — COMMENTED** [review]
you can make the changes suggested and we are good to go

**Comment by lovisgod** [comment]
@adaezeodurukwe you can merge the PR

**Review by adaezeodurukwe — APPROVED** [review]

**Pull request merged**